backend/app/Http/Controllers/AuthController.php

```diff
@@ -2,75 +2,56 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\LoginRequest;
+use App\Http\Requests\RegisterRequest;
+use App\Http\Requests\UpdateUserRequest;
+use App\Http\Resources\UserResource;
+use App\Models\User;
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
-use App\Models\User;
-use Illuminate\Support\Facades\Hash;
-use Illuminate\Validation\Rule;
 
 class AuthController extends Controller
 {
-    public function login(Request $request) {
-        $credentials = $request->validate([
-            'email' => ['required', 'email'],
-            'password' => ['required'],
-        ]);
-
-        if (Auth::attempt($credentials)) {
-            // $request->session()->regenerate();
+    public function login(LoginRequest $request)
+    {
+        $credentials = $request->validated();
 
+        if (!Auth::attempt($credentials)) {
             return response()->json([
-                'token' => auth()->user()->createToken('hello')->plainTextToken
-            ]);
+                'message' => 'The provided credentials do not match our records.'
+            ], 401);
         }
 
         return response()->json([
-            'message' => 'The provided credentials do not match our records.'
-        ], 400);
-    }
-
-    public function registration(Request $request) {
-        $validatedData = $request->validate([
-            'name' => 'required|string|max:255',
-            'email' => 'required|string|email|max:255|unique:users',
-            'password' => 'required|string|min:8',
-        ]);
-
-        $user = User::create([
-            'name' => $validatedData['name'],
-            'email' => $validatedData['email'],
-            'password' => Hash::make($validatedData['password']),
+            'token' => Auth::user()->createToken('authToken')->plainTextToken
         ]);
 
-        $token = $user->createToken('auth_token')->plainTextToken;
+    }
 
+    public function register(RegisterRequest $request)
+    {
+        $user = User::create($request->validated());
```

Comment:
what if user will not be registered? For some reason not related to fields validation

Reply:
do u want me to wrap all other funcs in try catch?

Reply:
Nope, you should check wether the user was registered or not. The previous logic was wrong because it will try to generate and send token in both positive and negative cases.